### PR TITLE
Palak Assignment 17_(30+20)50Min

### PR DESCRIPTION
**Pull request by PalakSJun20**
image
image

### CODE REVIEW

File: src/palakSJun20/CountCharacterFromString.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+				char ch = string.charAt(index);
+				if(Character.isDigit(ch))
+					digitCount++;
+				else if(Character.isUpperCase(ch)) {
+					upperCaseCount++;
+					totalCount++;
+				}
+				else if(Character.isLowerCase(ch)) {
+					lowerCaseCount++;
+					totalCount++;
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/CountCharacterFromString.java]: totalCount this variable is not needed, you can write logic without this variable too.

File: src/palakSJun20/FirstLastIndexOfCharacters.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	//Method for Finding first and last index of every characters
+	void findFirstAndLastIndex(String input){
+		
+		for (int index = 0; index < input.length(); index++) {
+			char ch = input.charAt(index);
+			int count = 0;
+				if(index == input.indexOf(ch)) {
+					for(int i = 0; i < input.length(); i++) {
+						if (input.charAt(i) == ch) {
+							count++;
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/FirstLastIndexOfCharacters.java:20]: You have increased the complexity in code, Using one for loop you can achieve the output.

File: src/palakSJun20/CountCharacterFromString.java (hunk 1 above)
Comment (PalakSJun20) [reply src/palakSJun20/CountCharacterFromString.java]: done
`(upperCaseCount+lowerCaseCount)`

File: src/palakSJun20/FirstLastIndexOfCharacters.java (hunk 2 above)
Comment (PalakSJun20) [reply src/palakSJun20/FirstLastIndexOfCharacters.java:20]: I m not sure it is correct way or not please check it took me more time to write new logic. :( 
didn't remove the previous method just commented method Calling and create new method.

File: src/palakSJun20/FirstLastIndexOfCharacters.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			}
+		}
+	}
+	
+	void FirstAndLast(String input){
+		for (int index = 0; index < input.length(); index++) {
+			char ch = input.charAt(index);
+			int count = 0;	
+			if( index == input.indexOf(ch) && input.indexOf(ch) < input.lastIndexOf(ch)  )
+				System.out.print(ch+ " -> First Index = "+ input.indexOf(ch) + " Last Index = "+ input.lastIndexOf(ch) +"\n");
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/FirstLastIndexOfCharacters.java:34]: Now this one is correct.

File: src/palakSJun20/FirstLastIndexOfCharacters.java (hunk 2 above)
Comment (KrishnaKTechnocredits) [reply src/palakSJun20/FirstLastIndexOfCharacters.java:20]: Np, It's completely fine.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**